### ISSUE TITLE
cannot rollback cilium from 1.13 to 1.12

### ISSUE THREAD

**Issue by sofat1989** [cilium/cilium#25473]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I upgraded the cilium to 1.13. And I tried to downgrade to 1.12. 
The node will become NotReady. 
All the packets are sent to this node will be dropped.

After flusing the iptables mangle table, the network on this node is restored.

I think the packets are dropped by the iptables version in the cilium. 

Cilium 1.13 (iptables v1.8.8 (nf_tables))
Cilium 1.12 (iptables v1.8.7 (nf_tables))

1. Deploy the cilium 1.13, cilium-agent will create these rules in the mangle table
```
# iptables -t mangle -S
...
-A CILIUM_PRE_mangle -p tcp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
```

2. Roll back cilium to 1.12. it will copy the rules. https://github.com/cilium/cilium/blob/main/pkg/datapath/iptables/iptables.go#L705 . Because its iptables' version is 1.8.7, it cannot get the mark flag (-m mark --mark 0x57840200). It will add these two rules below.
```
~# iptables -t mangle -S
...
-A CILIUM_PRE_mangle -p tcp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
``` 

3. Later it will add two new rules containing the mark (-m mark --mark 0x57840200). So finally there are 4 iptables rules in the mangle shown below
```
-A CILIUM_PRE_mangle -p tcp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p tcp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
```
All the packets will be redirected to tproxy and dropped.

This issue is caused by the iptables compatibility, iptables 1.8.7 cannot get the mark in the rules set by iptables 1.8.8

In the ubuntu 22.04, the iptables is 1.8.7. You can see the differences of the results between 1.8.7 and i.8.8
```
node:~#iptables --version
iptables v1.8.7 (nf_tables)
node:~# iptables -t mangle -S | grep cilium-dns-egress
-A CILIUM_PRE_mangle -p tcp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
node:~# crictl ps | grep agent
eaf3ea5b7e660       d5a34d8426949       21 minutes ago      Running             cilium-agent            0                   a565e26f7d6da       cilium-e2e-99rmt
node:~# crictl exec -it eaf3ea5b7e660 bash
clium:/home/cilium# iptables --version
iptables v1.8.8 (nf_tables)
clium:/home/cilium#  iptables -t mangle -S | grep cilium-dns-egress
-A CILIUM_PRE_mangle -p tcp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
-A CILIUM_PRE_mangle -p udp -m mark --mark 0x57840200 -m comment --comment "cilium: TPROXY to host cilium-dns-egress proxy" -j TPROXY --on-port 33879 --on-ip 0.0.0.0 --tproxy-mark 0x200/0xffffffff
```

### Cilium Version

1.13.1, 1.12.9

### Kernel Version

5.15.0-26-generic

### Kubernetes Version

v.1.18.8

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aditighag** [comment]
Hi! There've been some fixes around iptables versioning, and they could potentially be related. See https://github.com/cilium/cilium/commit/b599d6f55a69bc45f9ea2b87a0f9f9775eaed71a and https://github.com/cilium/cilium/commit/e364d068618ad881336212b8dbc96d2d481f1238. @jibi Do we document the supported iptable versions for Cilium?

**Comment by jibi** [comment]
@sofat1989 I'm afraid the only solution for the downgrade is, as you pointed out, manually deleting those rules, as Cilium v1.13 uses a newer iptables version which is not compatible (at least wrt those `-m mark` rules) with the 1.8.7 version shipped with Cilium v1.12. I'll add a note to the upgrade guide
